core/src/main/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbe.java

```diff
@@ -76,25 +87,62 @@ protected boolean requiresRelease() {
         return true;
     }
 
-    private ProbeResult fromSCMLink(ProbeContext context, String scm) {
+    private ProbeResult fromSCMLink(ProbeContext context, String scm, String pluginName) {
         Matcher matcher = GH_PATTERN.matcher(scm);
         if (!matcher.find()) {
             if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug("{} is not respecting the SCM URL Template", scm);
+                LOGGER.debug("{} is not respecting the SCM URL Template.", scm);
             }
-            return ProbeResult.failure(key(), "SCM link doesn't match GitHub plugin repositories");
+            return ProbeResult.failure(key(), "SCM link doesn't match GitHub plugin repositories.");
         }
-
         try {
             context.getGitHub().getRepository(matcher.group("repo"));
-            return ProbeResult.success(key(), "The plugin SCM link is valid");
+            Optional<Path> file = searchPomFiles(context.getScmRepository(), pluginName);
+            String folderPath = file.isPresent() ? new File(file.toString()).getParentFile().getName() : "";
```

Comment:
Am I properly understanding that in your code, it is valid for a repository to NOT contain a `pom.xml` for a plugin?

AFAICT, you SHOULD always find a `pom.xml` matching your plugin in the repository. Either it is at the root of the repository or not, but it should exist.

If it is not found, it means that the plugin you are searching for does not exist in your repository, and I feel like this Probe is not dealing with that situation.

Reply:
> AFAICT, you SHOULD always find a pom.xml matching your plugin in the repository. Either it is at the root of the repository or not, but it should exist.

we still have 80 plugins using Gradle as build tool.

The problem with gradle is that it's really complicated to parse the build description.

Reply:
Oooooh that is right! I totally forgot about that, thanks for the reminder @alecharp!

How do you consider those plugins for the other probes requiring Maven stuff btw?

Reply:
For the moment, with difficulties. It's a problem we know (we probably should advertise that a bit more).